game/src/main/kotlin/world/gregs/voidps/world/interact/entity/npc/shop/GeneralStore.kts

```diff
@@ -0,0 +1,60 @@
+package world.gregs.voidps.world.interact.entity.npc.shop
+
+import world.gregs.voidps.engine.entity.character.npc.npcOperate
+import world.gregs.voidps.world.interact.dialogue.Happy
+import world.gregs.voidps.world.interact.dialogue.Neutral
+import world.gregs.voidps.world.interact.dialogue.Talk
+import world.gregs.voidps.world.interact.dialogue.type.choice
+import world.gregs.voidps.world.interact.dialogue.type.npc
+import world.gregs.voidps.world.interact.dialogue.type.player
+import world.gregs.voidps.world.interact.entity.npc.shop.openShop
+
+npcOperate("Trade", "shopkeeper*", "shop_assistant*") {
+    player.openShop(target.def.getOrNull<String>("shop") ?: return@npcOperate)
+}
+
+npcOperate("Talk-to", "shopkeeper*") {
+    if (target.id.endsWith("musa_point")) {
+        npc<Happy>("Can I help you at all?")
+        //todo find right dialogue from A_Void_Dance quest?
+    } else {
+        npc<Happy>("Can I help you at all?")
+    }
```

Comment:
```suggestion
    // TODO find right dialogue from A_Void_Dance quest?
```

Reply:
i removed it it can be added in when someone dose the quest